GPUOffload/GPUOffload.hh

```diff
@@ -2,7 +2,27 @@
 
 #include "GPUOffloadInterface.hh"
 
-// Offloader instance is thread local (for Celeritas only at present)
-// NB: Choosing between AdePT/Celeritas might be tricky because it's required
-// that the returned reference is thread-local.
-GPUOffloadInterface& GPUOffload(/* argument to choose implementation ?*/);
+//! Available offload backends
+enum class GPUOffloadBackend
+{
+  Adept,
+  Celeritas,
+  None,
+  Unknown
+};
+
+//! Options for type of offload backend and its parameters
+struct GPUOffloadOptions
+{
+  //! Implementation to use for offload backend
+  GPUOffloadBackend backend = GPUOffloadBackend::Unknown;
+
+  //! True if known backend selected
+  explicit operator bool() const { return backend != GPUOffloadBackend::Unknown; }
+};
+
+//! Setup offload backend from input options
+void GPUOffloadSetup(const GPUOffloadOptions& opts);
+
+//! Get reference to offloader for the current thread
+GPUOffloadInterface& GPUOffload();
```

Comment:
This may not be the right place to discuss (perhaps I should make an issue?), but one of the requests of the R&D assessment was to support CPU offloading too: it improves the ability to test, deploy, etc.; and we may even get a performance boost from the improved algorithms/memory layout/etc.

So should we start building in the ability to support *CPU* offload as well, and if so what should we name this? Should it just be for EM tracks on GPU, or do we make it more general?

Reply:
Good point, I've started #10 to follow up on this. I think it's straightforward, but feedback welcome!